### ISSUE TITLE
Shuffle instance ordering within a frame during training

### ISSUE THREAD

**Issue by talmo** [talmolab/biogtr#31]
In an exploration, we found that transformer seems to be sensitive to the ordering of the instances within the frame at inference time, sometimes leading to very different results.

Example association matrices:

```py
asso_output, embed = model(frames[:2], query_frame=1)
asso_output[0].cpu().detach().numpy()
```
```
array([[2.008031  , 0.01795167, 2.0095046 , 0.01274773],
       [2.0929549 , 2.9742773 , 2.094697  , 2.9708583 ]], dtype=float32)
```

Then after flipping the instance ordering:

```py
frames[0].instances = frames[0].instances[::-1]
frames[1].instances = frames[1].instances[::-1]
asso_output, embed = model(frames[:2], query_frame=1)
asso_output[0].cpu().detach().numpy()
```
```
array([[4.1108932, 2.613935 , 4.1106267, 2.6162283],
       [2.648179 , 4.704933 , 2.639312 , 4.704572 ]], dtype=float32)
```

Note that flipping the order back gets back to the same association matrix, but it's concerning that it's so sensitive to the instance ordering since that's intended to be arbitrary.

A bug in how we handle embeddings and such notwithstanding, then the next best thing is to at least shuffle the instance order within frames during training to break any potential associations deriving from the input data.

~~For example, in one of our mouse datasets, we always have the black mouse as the first track, and so would appear as the first instance.~~ Possibly, it can emerge out of an imbalance? For a sample session, ~73% of track 0 instances were also the first instance in the frame.

**Comment by aaprasad** [comment]
potentially save the instance index in `Instance` object for bookkeeping so we can do diagnostics on distribution of shuffled index. Especially when theres only two instances shuffling may not change the order as frequently as we'd hope so it could be useful to have that info

**Comment by aaprasad** [comment]
done in #38
